src/app/fdctl/monitor/monitor.c

```diff
@@ -184,116 +151,157 @@ run_monitor( config_t * const config,
+        break;
+      case wksp_bank:
+        tiles[ tile_idx ].name = "bank";
+        tiles[ tile_idx ].cnc = fd_cnc_join( fd_wksp_pod_map( pod, "cnc" ) );
+        if( FD_UNLIKELY( !tiles[ tile_idx ].cnc ) ) FD_LOG_ERR(( "fd_cnc_join failed" ));
+        if( FD_UNLIKELY( fd_cnc_app_sz( tiles[ tile_idx ].cnc )<64UL ) ) FD_LOG_ERR(( "cnc app sz should be at least 64 bytes" ));
         tile_idx++;
         break;
     }
   }
 
+  FD_TEST( tile_idx == tile_cnt );
+  FD_TEST( link_idx == link_cnt );
+
   /* Setup local objects used by this app */
   fd_rng_t _rng[1];
   fd_rng_t * rng = fd_rng_join( fd_rng_new( _rng, seed, 0UL ) );
 
-  snap_t * snap_prv = (snap_t *)fd_alloca( alignof(snap_t), sizeof(snap_t)*2UL*tile_cnt );
-  if( FD_UNLIKELY( !snap_prv ) ) FD_LOG_ERR(( "fd_alloca failed" )); /* Paranoia */
-  snap_t * snap_cur = snap_prv + tile_cnt;
+  tile_snap_t * tile_snap_prv = (tile_snap_t *)fd_alloca( alignof(tile_snap_t), sizeof(tile_snap_t)*2UL*tile_cnt );
+  if( FD_UNLIKELY( !tile_snap_prv ) ) FD_LOG_ERR(( "fd_alloca failed" )); /* Paranoia */
+  tile_snap_t * tile_snap_cur = tile_snap_prv + tile_cnt;
+
+  link_snap_t * link_snap_prv = (link_snap_t *)fd_alloca( alignof(link_snap_t), sizeof(link_snap_t)*2UL*link_cnt );
+  if( FD_UNLIKELY( !link_snap_prv ) ) FD_LOG_ERR(( "fd_alloca failed" )); /* Paranoia */
+  link_snap_t * link_snap_cur = link_snap_prv + link_cnt;
 
   /* Get the inital reference diagnostic snapshot */
-  snap( tile_cnt, snap_prv, tile_cnc, tile_mcache, tile_fseq );
+  tile_snap( tile_snap_prv, tile_cnt, tiles );
+  link_snap( link_snap_prv, link_cnt, links );
   long then; long tic; fd_tempo_observe_pair( &then, &tic );
 
   /* Monitor for duration ns.  Note that for duration==0, this
      will still do exactly one pretty print. */
   FD_LOG_NOTICE(( "monitoring --dt-min %li ns, --dt-max %li ns, --duration %li ns, --seed %u", dt_min, dt_max, duration, seed ));
 
-  /* Setup TTY */
-  printf( TEXT_CUP_HOME );
-
   long stop = then + duration;
   if( duration == 0) stop = LONG_MAX;
 
+#define PRINT( ... ) do {                                                        \
+    int n = snprintf( *buf, *buf_sz, __VA_ARGS__ );                              \
+    if( FD_UNLIKELY( n<0 ) ) FD_LOG_ERR(( "snprintf failed" ));                  \
+    if( FD_UNLIKELY( (ulong)n>=*buf_sz ) ) FD_LOG_ERR(( "snprintf truncated" )); \
+    *buf += n; *buf_sz -= (ulong)n;                                              \
+  } while(0)
+
+  write_stdout( TEXT_ALTBUF_ENABLE, sizeof(TEXT_ALTBUF_ENABLE)-1 );
+  write_stdout( TEXT_NOCURSOR, sizeof(TEXT_NOCURSOR)-1 );
+
   for(;;) {
     /* Wait a somewhat randomized amount and then make a diagnostic
        snapshot */
     fd_log_wait_until( then + dt_min + (long)fd_rng_ulong_roll( rng, 1UL+(ulong)(dt_max-dt_min) ) );
 
-    snap( tile_cnt, snap_cur, tile_cnc, tile_mcache, tile_fseq );
+    tile_snap( tile_snap_cur, tile_cnt, tiles );
+    link_snap( link_snap_prv, link_cnt, links );
```

Comment:
This should be `link_snap_cur`, right?

Reply:
Good catch! Was wondering why it flickered on boot